app/src/main/kotlin/com/google/samples/apps/nowinandroid/navigation/NiaNavigatorImpl.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.navigation
+
+import com.google.samples.apps.nowinandroid.core.navigation.NiaBaseNavigator
+import com.google.samples.apps.nowinandroid.core.navigation.NiaNavigator
+import javax.inject.Inject
+
+internal class NiaNavigatorImpl @Inject constructor(
+    // Key should be the Class for the navigator, value should be the navigator instance itself
+    private val navigators: Map<@JvmSuppressWildcards Class<*>, @JvmSuppressWildcards NiaBaseNavigator<*, *, *>>,
+): NiaNavigator {
+    override fun <T : NiaBaseNavigator<out Any, out Any, out Any>> getNavigator(clazz: Class<T>): T {
+        return navigators[clazz] as T
```

Comment:
This will throw at runtime if we try to get a navigator without providing a multibinding for it.